src/apps/api/views/leaderboards.py

```diff
@@ -45,13 +48,13 @@ def update(self, request, *args, **kwargs):
 def add_submission_to_leaderboard(request, submission_pk):
     # TODO: rebuild this to look somewhere else for what leaderboard to post to?
     submission = get_object_or_404(Submission, pk=submission_pk)
-    competition = submission.phase.competition
+    phase = submission.phase
 
     # Removing any existing submissions on leaderboard
-    Submission.objects.filter(phase__competition=competition, owner=request.user).update(leaderboard=None)
+    Submission.objects.filter(phase=phase, owner=request.user).update(leaderboard=None)
 
-    # Assume that submission.scores.first().column.leaderboard will always have the correct leaderboard
-    leaderboard = submission.scores.first().column.leaderboard
+    leaderboard = submission.phase.leaderboard
+    # Assume that submission.children.first().scores.first().column.leaderboard will always have the correct leaderboard
```

Comment:
phase.leaderboard should have the same result as submission.phase.leaderboard. Also, Is that comment out of date? We should delete it or change it.